### PR TITLE
[RFC] Add Hyperbackup support

### PR DESCRIPTION
**Pull request by catchdave**
Hey @mib1185, 

**Request For Comment**

I'd like to incorporate support for hyperbackup so that I can have the backup tasks show up in home assistant. This code started with the work from this [synology api repo](https://github.com/N4S4/synology-api) and then augmented by reverse engineering from the javascript files/API on my NAS, so it is not complete, but covers many use cases so far.

What needs to be done to get this PR incorporated? Obviously there are missing unit tests - but what else?

### CODE REVIEW

File: src/synology_dsm/api/hyperbackup/hyperbackup.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            else:
+                return STATUS_ERROR
+        elif raw_status in [PROP_STATUS_BACKUP, PROP_STATUS_DETECT, PROP_STATUS_VER_DEL, PROP_STATUS_PREP_VER_DEL]:
+            return STATUS_RESUMING if previous_result == RESULT_RESUME else STATUS_RUNNING
+        elif raw_status == PROP_STATUS_WAITING:
+            return STATUS_WAITING
+        else:
+            return STATUS_ERROR
+
+    def name(self, task_id: int) -> str:
```
[the remark this answers is not in the document]
Comment (mib1185) [reply src/synology_dsm/api/hyperbackup/hyperbackup.py:141]: maybe it is worth to define a class for tasks (eq. `SynoHyperbackupTask`), which has only the needed/wanted proeprties/attributes (_similar to the [`SynoCamera`](https://github.com/mib1185/py-synologydsm-api/blob/db2c8f196cc79d14e34ed5d44ec8bafe244e5ef5/src/synology_dsm/api/surveillance_station/camera.py#L5) class_)

File: src/synology_dsm/synology_dsm.py [hunk 2]
```diff
@@ -364,6 +369,9 @@ def reset(self, api: any) -> bool:
             if hasattr(self, "_" + api):
                 setattr(self, "_" + api, None)
                 return True
+            if api == SynoHyperBackup.API_KEY:
```
Comment (mib1185) [inline src/synology_dsm/synology_dsm.py:372]: please keep this "list" sorted

File: src/synology_dsm/synology_dsm.py [hunk 3]
```diff
@@ -388,6 +396,9 @@ def reset(self, api: any) -> bool:
             if api == SynoSurveillanceStation.API_KEY:
                 self._surveillance = None
                 return True
+        if isinstance(api, SynoHyperBackup):
```
Comment (mib1185) [inline src/synology_dsm/synology_dsm.py:399]: please keep this "list" sorted

File: src/synology_dsm/synology_dsm.py [hunk 4]
```diff
@@ -414,6 +425,13 @@ def reset(self, api: any) -> bool:
             return True
         return False
 
+    @property
+    def hyperbackup(self) -> SynoHyperBackup:
```
Comment (mib1185) [inline src/synology_dsm/synology_dsm.py:429]: please keep this "list" sorted

File: src/synology_dsm/synology_dsm.py [hunk 5]
```diff
@@ -72,6 +73,7 @@ def __init__(
         self._apis = {
             "SYNO.API.Info": {"maxVersion": 1, "minVersion": 1, "path": "query.cgi"}
         }
+        self._hyperbackup = None
```
Comment (mib1185) [inline src/synology_dsm/synology_dsm.py:76]: please keep this "list" sorted

File: src/synology_dsm/synology_dsm.py [hunk 6]
```diff
@@ -326,6 +328,9 @@ async def _execute_request(self, method: str, url: str, params: dict, **kwargs):
 
     async def update(self, with_information: bool = False, with_network: bool = False):
         """Updates the various instanced modules."""
+        if self._hyperbackup:
```
Comment (mib1185) [inline src/synology_dsm/synology_dsm.py:331]: please keep this "list" sorted

**Review by mib1185 — CHANGES_REQUESTED** [review]
Hi @catchdave

many thanks for this contribution.
i tried already in the past to add the hyperbackup feature (#31), but this somehow staled out 🙈 
so it is highly appreciated to see this happen now 🙂 

Yes, unit tests are needed (_maybe you can recycle some of them from my previous try in #31_).
Further i've aded already some comments and suggestions, but tbh did not test the code yet.
Last but not least, ~~you may want to have a look into the [CONTRIBUTING](https://github.com/mib1185/py-synologydsm-api/blob/master/CONTRIBUTING.rst) guide~~ i see i've to urgently update the [CONTRIBUTING](https://github.com/mib1185/py-synologydsm-api/blob/master/CONTRIBUTING.rst) guide 🙈 
TLDR; if possible, i would highly recommend to use the dev-container in this repository, which integrates multiple linting and formatting automation (_"forced" by pre-commit hook_)

So please let us work together to bring this on the road 👍 

regards,
Michael
